src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -868,33 +882,122 @@ bool HMICapabilitiesImpl::rc_supported() const {
+ * @param name_of_member name which we should be checked
+ * @returns true if member exists, otherwise returns false
+ */
+bool check_existing_json_member(const Json::Value& json_member,
+                                const char* name_of_member) {
+  return !json_member.isNull() && json_member.isMember(name_of_member);
+}
+
+/**
+ * @brief Helper class for obtaining proper capabilities JSON value considering
+ * case if it were overriden by cache
+ */
+struct JsonCapabilitiesGetter {
+ public:
+  /**
+   * @brief JsonCapabilitiesGetter constructor
+   * @param json_main_node reference to the main JSON capabilities node
+   * @param json_override_node reference to overriden JSON capabilities node
+   */
+  JsonCapabilitiesGetter(Json::Value& json_main_node,
+                         Json::Value& json_override_node)
+      : json_main_node_(json_main_node)
+      , json_override_node_(json_override_node) {}
+
+  Json::Value GetInterfaceJsonMember(const char* member_name) {
+    if (check_existing_json_member(json_override_node_, member_name)) {
+      return GetOverrideJsonMember(member_name);
+    }
+
+    if (check_existing_json_member(json_main_node_, member_name)) {
+      return GetMainJsonMember(member_name);
+    }
+
+    return Json::Value::null;
+  }
+
+  /**
+   * @brief GetJsonMember gets JSON value for a specified JSON member from
+   * overriden JSON node if member exists, otherwise main JSON node will be
+   * taken
+   * @param member_name name of the JSON member to get
+   * @return JSON value for specified member or Value::null if not found
+   */
+  Json::Value GetJsonMember(
+      const char* member_name,
+      hmi_apis::FunctionID::eType request_id,
+      std::vector<hmi_apis::FunctionID::eType>& interfaces_to_update) {
+    if (check_existing_json_member(json_override_node_, member_name)) {
+      return GetOverrideJsonMember(member_name);
+    }
+
+    if (check_existing_json_member(json_main_node_, member_name)) {
+      interfaces_to_update.push_back(request_id);
+      return GetMainJsonMember(member_name);
+    }
+
+    return Json::Value::null;
+  }
+
+  /**
+   * @brief GetMainJsonMember gets JSON value for a specified JSON member from
+   * the main JSON node
+   * @param member_name name of the JSON member to get
+   * @return JSON value for specified member or Value::null if not found
+   */
+  Json::Value GetMainJsonMember(const char* member_name) {
+    return json_main_node_.get(member_name, Json::Value::null);
+  }
+
+  /**
+   * @brief GetOverrideJsonMember gets JSON value for a specified JSON member
+   * from the overriden JSON node
+   * @param member_name name of the JSON member to get
+   * @return JSON value for specified member or Value::null if not found
+   */
+  Json::Value GetOverrideJsonMember(const char* member_name) {
+    return json_override_node_.get(member_name, Json::Value::null);
+  }
+
+ private:
+  Json::Value& json_main_node_;
+  Json::Value& json_override_node_;
```

Comment:
@LitvinenkoIra I would like to propose to changes the names to the next examples:
`json_default` and `json_cache`